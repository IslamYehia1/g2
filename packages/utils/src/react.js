import React from 'react';
import { is } from './is';

export function getValidChildren(children) {
	return React.Children.toArray(children).filter((child) =>
		React.isValidElement(child),
	);
}

export function assignRef(ref, value) {
	if (ref == null) return;

	if (is.function(ref)) {
		ref(value);
		return;
	}

	try {
		//@ts-ignore
		ref.current = value;
	} catch (error) {
		throw new Error(`Cannot assign value '${value}' to ref '${ref}'`);
	}
}

export function mergeRefs(...refs) {
	return (value) => {
		refs.forEach((ref) => assignRef(ref, value));
	};
}
