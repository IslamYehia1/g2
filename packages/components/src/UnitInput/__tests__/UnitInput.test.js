import { render } from '@testing-library/react';
import React from 'react';

import { UnitInput } from '../index';

describe('props', () => {
	test('should render correctly', () => {
		const { container } = render(<UnitInput id="text-input" />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
