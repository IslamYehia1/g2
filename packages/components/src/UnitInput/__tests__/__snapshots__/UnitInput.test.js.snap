// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render correctly 1`] = `
<div
  class="wp-components-flex wp-components-base-field wp-components-text-input wp-components-unit-input css-in42v"
  data-g2-c16t="true"
  data-g2-component="TextInputWrapper"
>
  <input
    autocomplete="off"
    class="css-16e6wng"
    data-g2-c16t="true"
    data-g2-component="TextInput"
    id="text-input"
    spellcheck="false"
    type="text"
    value=""
  />
  <div
    class="wp-components-flex wp-components-h-stack wp-components-v-stack css-bv5fib"
    data-g2-c16t="true"
    data-g2-component="TextInputArrows"
  >
    <div
      class="wp-components-icon css-1vrpdm9"
      data-g2-c16t="true"
      data-g2-component="TextInputArrowUp"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 20 20"
      >
        <path
          d="M15 12l-5-5-5 5h10z"
        />
      </svg>
    </div>
    <div
      class="wp-components-icon css-7v1vli"
      data-g2-c16t="true"
      data-g2-component="TextInputArrowDown"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 20 20"
      >
        <path
          d="M5 8l5 5 5-5H5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
